opencog/attention/experimental/HebbianUpdatingAgent.cc

```diff
@@ -52,7 +57,12 @@ void HebbianUpdatingAgent::run()
     std::back_insert_iterator< std::vector<Handle> > out_hi(atoms);
 
     _as->get_handle_set_in_attentional_focus(out_hi);
-
+     /*Without adding this sleep code right below the above method call,
+     * nlp-parse evaluation thread waits for minutes before it gets a chance to
+     * run.
+     */
+    std::this_thread::sleep_for(std::chrono::nanoseconds(1));
```

Comment:
I have observed the same behavior even when running only one of these ECAN agents together with the NLP threads. It only happens when the _as->get_handle_set_in_attentional_focus is called, which ends up in method [ImportanceIndex::getHandleSet] (https://github.com/opencog/atomspace/blob/master/opencog/atomspace/ImportanceIndex.cc#L71). Seems there is a lot of copying going on there but I am not sure how exactly it made the whole thing slow.